### DISCUSSION TITLE
Bedrock provider when using llamaindex

### DISCUSSION THREAD

**Discussion by mdciri** [langfuse/langfuse#2438] [Ideas]
### Describe the feature or potential improvement

I am using AWS Bedrock and llama-index for my RAG. I would like that langfuse could trace asynchronous function as `get_responses`, `aget_responses`, or `batch_runner.aevaluate_responses` in [example](https://docs.llamaindex.ai/en/stable/examples/retrievers/auto_merging_retriever/#evaluation).

When I do with asynchronous function, I do not get a nice overview of the trace as when I do `engine.query(my_query)`.

Moreover, when I do `engine.query(my_query)` I get in the LLM bedrock generation ObservationView with the input and output number of tokens in the `observtion.output`, but nothing in `observation.usage` with also the cost of the call to the service.

**Comment by hassiebp** [comment]
Hi @mdciri, thanks for your report! As for the issues you described:
- The LlamaIndex callback integration that Langfuse relies upon is not entirely concurrency-safe. Thus, the trace you get for your asynchronous execution might be erroneous. This will hopefully be mitigated once we move the the LlamaIndex `instrumentation` module to power Langfuse observability.
- As for the token in the `output` instead of `usage`, could you please share a minimum repro example for me to take a look at?
